### DISCUSSION TITLE
Stop Argo from adding random string to workflow name?

### DISCUSSION THREAD

**Discussion by hyler** [argoproj/argo-workflows#9331] [Q&A] [answered]
Good morning,

We're PoCing Argo, but one thing I noticed with the example deployment is that when I `apply` a workflow, it shows in the UI with a string of random characters appended to it. Is there a way to stop this?

The reason this is important is because we want to be able to update a workflow by `apply`ing it again, with the same name—the intent being that the new YAML will overwrite the old workflow, instead of having a new one.

That's how we do it in Airflow (which we are dying to replace) right now—we update a job config, register it with Airflow, and it picks up the changes on the next run. The name of the job stays the same.

**Comment by terrytangyuan** [comment, marked as answer]
Use `name` instead of `generateName`.
